src/plone/api/user.py

```diff
@@ -232,7 +232,11 @@ def get_roles(username=None, user=None, obj=None, inherit=True):
         else:
             username = user.getId()
 
-    user = portal_membership.getMemberById(username)
+    if username is None and user is None:
+        user = portal_membership.getAuthenticatedMember()
+    else:
+        user = portal_membership.getMemberById(username)
```

Comment:
Correct. I tried refactoring this, but then somehow tests for get_roles_in_context start failing.